src/bosh-director/lib/bosh/director/api/controllers/deployments_controller.rb

```diff
@@ -595,7 +595,7 @@ def create_vms_response(vms_instances_hash)
         results = []
         vms_instances_hash.each_pair do |instance, vms|
           vms.each do |vm|
-            results << create_vm_response(instance, vm).merge('active' => vm.active)
+            results << create_vm_response(instance, vm).merge('active' => vm.active).merge('permanent_nats_credentials' => vm.permanent_nats_credentials)
```

Comment:
This will make the `permanent_nats_credentials` attribute come back as part of a `/deployments/NAME/vms` call. If it it was instead added to the `create_vm_response` method directly, rather than being merged in here, it would also come back as part of `/deployments/NAME/instances` api calls.

I'm not really sure it needs to be in both, but it doesn't appear like it would be harmful to include in both.

Did you have a better understanding of this than me and only include it in `vms` for a particular reason?

Reply:
@jpalermo the rationale to add it in the vms endpoint because the parameter was added to the vms table in the database, and is not part of the instances table. Also in the vms table we have another parameter that is the nats_config_sha so it made sense to leave the nats stuff on the vms call. Also nats-sync uses the vms endpoint.

Reply:
If you look at `create_vm_response` about half the response data already comes from the VMs table, so the `/instances` call is already polluted quite a bit. But I can't think of any real reason to add it to both, and the cost of doing that in the future if we do end up needing it there is pretty low, so it's probably fine to leave it just on `vms`